include/control_filters/low_pass_filter.hpp

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2023, Stogl Robotics Consulting UG (haftungsbeschränkt)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef CONTROL_FILTERS__LOW_PASS_FILTER_HPP_
+#define CONTROL_FILTERS__LOW_PASS_FILTER_HPP_
+
+#include <Eigen/Dense>
+#include <cmath>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "low_pass_filter_parameters.hpp"
+#include "filters/filter_base.hpp"
+
+#include "geometry_msgs/msg/wrench_stamped.hpp"
+
+namespace control_filters
+{
+
+template <typename T>
+class LowPassFilter : public filters::FilterBase<T>
+{
+public:
+  LowPassFilter();
+
+  ~LowPassFilter() override;
+
+  bool configure() override;
+
+  bool update(const T & data_in, T & data_out) override;
+
+protected:
+  void compute_internal_params()
+  {
+    a1_ = exp(
+      -1.0 / parameters_.sampling_frequency * (2.0 * M_PI * parameters_.damping_frequency) /
+      (pow(10.0, parameters_.damping_intensity / -10.0)));
+    b1_ = 1.0 - a1_;
+  };
+
+private:
+  rclcpp::Clock::SharedPtr clock_;
+  std::shared_ptr<rclcpp::Logger> logger_;
+  std::shared_ptr<low_pass_filter::ParamListener> parameter_handler_;
+  low_pass_filter::Params parameters_;
+
+  // Filter parameters
+  // TODO(destogl): we should do this more intelligently using only one set of types
+  double filtered_value, filtered_old_value, old_value;
+  Eigen::Matrix<double, 6, 1> msg_filtered, msg_filtered_old, msg_old;
+  double a1_;
+  double b1_;
+};
+
+template <typename T>
+LowPassFilter<T>::LowPassFilter() : a1_(1.0), b1_(0.0)
+{
+}
+
+template <typename T>
+LowPassFilter<T>::~LowPassFilter()
+{
+}
+
+template <typename T>
+bool LowPassFilter<T>::configure()
+{
+  clock_ = std::make_shared<rclcpp::Clock>(RCL_SYSTEM_TIME);
+  logger_.reset(
+    new rclcpp::Logger(this->logging_interface_->get_logger().get_child(this->filter_name_)));
+
+  // Initialize the parameters once
+  if (!parameter_handler_)
+  {
+    try
+    {
+      parameter_handler_ =
+        std::make_shared<low_pass_filter::ParamListener>(this->params_interface_, this->param_prefix_);
```

Comment:
this->param_prefix_ was missing but was not detected by the current tests because one just loads and instantiates the filter with no configure, and the second test configures with local parameters.
I suppose a rostest with param loading from yaml is necessary to ensure loading the filters is feasible within a namespace.